django/db/backends/postgresql/base.py

```diff
@@ -359,26 +343,50 @@ def get_new_connection(self, conn_params):
             )
         return connection
 
+    @cached_property
+    def role_name(self):
+        return self.settings_dict["OPTIONS"].get("assume_role")
+
+    @classmethod
+    def init_connection_role(cls, connection, role_name=None):
+        if role_name:
+            ops = cls.ops_class(connection)
```

Comment:
`ops_class` takes a `BaseDatabaseWrapper` and not the actual psycopg connection -- or did I get confused with all the wrapping now?

Reply:
Interesting
You're right that `ops_class` takes a `BaseDatabaseWrapper` in the init (`self.ops = self.ops_class(self)`), but `BaseDatabaseOperations` takes a "connection" in it's init, and so I am not sure, it looks like it can take both

Reply:
I think I'm wrong, I'll do an update